src/main/kotlin/com/msd/robot/domain/Robot.kt

```diff
@@ -174,4 +180,15 @@ class Robot(
     fun totalUpgrades(): Int {
         return damageLevel + energyLevel + energyRegenLevel + healthLevel + inventory.storageLevel + miningSpeedLevel + miningLevel
     }
+
+    /**
+     * Regenerates this [Robot's] [Robot] `energy`. Energy Regeneration is doubled when the Robot is in the [Player's][Player] spawn
+     */
+    fun regenerateEnergy() {
+        energy += if (planet.type == PlanetType.SPAWN) {
```

Comment:
The Spawn also has to be the spawn of the robots owner, not just any spawn. Then also test it

Reply:
A robot shouldn't be able to enter a spawn that doesn't belong to them, therfore testing for it here is unnecesarry.

Reply:
We shouldn't rely on other features to work, if for some reason a robot ends up on the wrong spawn, it at least shouldn't regenerate with twice the rate

Reply:
Why should we check something that we already check in other place to make sure it can't happen? That's just unneccessary code duplication.